core/src/HIP/Kokkos_HIP_ReduceScan.hpp

```diff
@@ -395,20 +387,11 @@ __device__ bool hip_single_inter_block_reduce_scan_impl(
   // Contributing blocks note that their contribution has been completed via an
   // atomic-increment flag If this block is not the last block to contribute to
   // this group then the block is done.
-  // FIXME_HIP __syncthreads_or is not supported by HIP yet.
-  // const bool is_last_block = !__syncthreads_or(
-  //    threadIdx.y
-  //        ? 0
-  //        : (1 + atomicInc(global_flags, block_count - 1) < block_count));
-  __shared__ int n_done;
-  n_done = 0;
-  __syncthreads();
-  if (threadIdx.y == 0) {
-    n_done = 1 + atomicInc(global_flags, block_count - 1);
-  }
-  __syncthreads();
-  bool const is_last_block = (n_done == static_cast<int>(block_count));
-
+  int n_done = 0;
+  const bool is_last_block = !__syncthreads_or(
+     threadIdx.y
+         ? 0
+         : (1 + atomicInc(global_flags, block_count - 1) < block_count));
```

Comment:
Sounds like a convoluted way to do `__syncthreads_and(threadIdx.y ? 1 : (atomicInc(global_flags, block_count - 1)  == block_count - 1));`

Reply:
heh, I blame our comment from years ago.  IIRC, this is how cuda does it as well: https://github.com/kokkos/kokkos/blob/80dd6adad7b9333eadd318dc8f98b27a3fe4d49c/core/src/Cuda/Kokkos_Cuda_ReduceScan.hpp#L611

Reply:
Yes I saw that.  Do you agree the version I suggested is more readable though?

Reply:
yes, agreed.